src/windows/minidump_writer.rs

```diff
@@ -0,0 +1,279 @@
+            Some(md::MINIDUMP_EXCEPTION_INFORMATION {
+                ThreadId: self.crash_context.thread_id,
+                // This is a mut pointer for some reason...I don't _think_ it is
+                // actually mut in practice...?
+                ExceptionPointers: self.crash_context.exception_pointers as *mut _,
+                ClientPointers: if self.is_external_process { 1 } else { 0 },
+            })
+        } else {
+            None
+        };
+
+        // This is a bit dangerous if doing in-process dumping, but that's not
+        // (currently) a real target of this crate, so this allocation is fine
+        let mut user_streams = Vec::with_capacity(2);
+
+        // Add an MDRawBreakpadInfo stream to the minidump, to provide additional
+        // information about the exception handler to the Breakpad processor.
+        // The information will help the processor determine which threads are
+        // relevant. The Breakpad processor does not require this information but
+        // can function better with Breakpad-generated dumps when it is present.
+        // The native debugger is not harmed by the presence of this information.
+        //
+        // This info is only relevant for in-process dumping
+        let mut breakpad_info = [0u8; 12];
+        if !self.is_external_process {
+            let bp_info = MINIDUMP_BREAKPAD_INFO {
+                validity: BreakpadInfoValid::DumpThreadId.bits()
+                    | BreakpadInfoValid::RequestingThreadId.bits(),
+                dump_thread_id: self.crash_context.thread_id,
+                // Safety: syscall
+                requesting_thread_id: unsafe { GetCurrentThreadId() },
+            };
+
+            // TODO: derive Pwrite for MINIDUMP_BREAKPAD_INFO
+            // https://github.com/rust-minidump/rust-minidump/pull/534
+            let mut offset = 0;
+            offset += breakpad_info.pwrite(bp_info.validity, offset)?;
+            offset += breakpad_info.pwrite(bp_info.dump_thread_id, offset)?;
+            breakpad_info.pwrite(bp_info.requesting_thread_id, offset)?;
+
+            user_streams.push(md::MINIDUMP_USER_STREAM {
+                Type: MINIDUMP_STREAM_TYPE::BreakpadInfoStream as u32,
+                BufferSize: breakpad_info.len() as u32,
+                // Again with the mut pointer
+                Buffer: breakpad_info.as_mut_ptr().cast(),
+            });
+        }
+
+        let handle_stream = self.fill_handle_stream();
+
+        // Note that we do this by ref, as the buffer inside the option needs
+        // to stay alive for as long as we're writing the minidump since
+        // the user stream has a pointer to it
+        if let Some((_buf, handle_stream)) = &handle_stream {
+            user_streams.push(*handle_stream);
+        }
+
+        let user_stream = md::MINIDUMP_USER_STREAM_INFORMATION {
```

Comment:
Perhaps `user_stream_info` would be a bit more clear of a name? Seems a bit strange that `user_stream` contains `user_streams` and not the other way around